mne/channels/tests/test_standard_montage.py

```diff
@@ -0,0 +1,258 @@
+@patch("mne.channels.DigMontage.__eq__",
+       _compare_dig_montage_and_standard_montage_)
+def test_read_montage(kind):
+    """Test difference between old and new standard montages."""
+    old_montage = read_montage(kind)
+    new_montage = read_standard_montage(kind)
+    assert new_montage == old_montage
+
+    raw = RawArray(
+        data=np.empty((len(old_montage.ch_names), 1), dtype=np.float64),
+        info=create_info(
+            ch_names=old_montage.ch_names, sfreq=1., ch_types='eeg'
+        )
+    ).set_montage(old_montage)
+
+    dig = raw.info['dig']
+    print(dig)
+    # pass
+
+from pytest import approx
+
+
+
+def test_read_standard_montage_egi_256():
+    """Test egi_256."""
+    EXPECTED_HEAD_SIZE = 0.085
+    EXPECTED_HEAD_VARIANCE = 0.00418
+    EXPECTED_FIRST_9_LOC = np.array(
+        [[ 6.55992516e-02,  5.64176352e-02, -2.57662946e-02],  # noqa
+         [ 6.08331388e-02,  6.57063949e-02, -6.40717015e-03],  # noqa
+         [ 5.19851171e-02,  7.15413471e-02,  1.12091555e-02],  # noqa
+         [ 4.18066179e-02,  7.31439438e-02,  2.66373224e-02],  # noqa
+         [ 3.09755787e-02,  6.97928339e-02,  4.21906579e-02],  # noqa
+         [ 1.96959622e-02,  6.22758709e-02,  5.58500821e-02],  # noqa
+         [ 1.03933314e-02,  5.14631908e-02,  6.63221724e-02],  # noqa
+         [ 8.76671630e-18,  3.81400691e-02,  7.39613137e-02],  # noqa
+         [-1.05002738e-02,  1.95003515e-02,  7.85765571e-02]]  # noqa
+    )
+
+    montage = read_standard_montage('EGI_256')
+    eeg_loc = np.array([ch['r'] for ch in _get_dig_eeg(montage.dig)])
+    eeg_center = eeg_loc.mean(axis=0)
+    distance_to_center = np.linalg.norm(eeg_loc - eeg_center, axis=1)
+
+    assert_allclose(eeg_center, [0, 0, 0], atol=1e-8)
+    assert_allclose(distance_to_center.mean(), 0.085, atol=1e-4)
+    assert_allclose(distance_to_center.std(), 0.00418, atol=1e-4)
+    # assert_allclose(eeg_loc[:9], EXPECTED_FIRST_9_LOC, atol=1e-1)  # XXX ?
+
+
+@pytest.mark.parametrize('kind', [
+    # 'EGI_256',
+    'easycap-M1',
+    # 'easycap-M10'
+])
+def test_foo(kind):
+    """Test difference between old and new standard montages."""
+    # import pdb; pdb.set_trace()
+    old_montage = read_montage(kind)
+    new_montage = read_standard_montage(kind)
+    # assert new_montage == old_montage
+
+
+def test_easycaps_are_indeed_different():
+    """Test something fishy with easycaps."""
+    m1 = read_montage('easycap-M1')
+    m10 = read_montage('easycap-M10')
+
+    assert m1.__repr__() == (
+        "<Montage | easycap-M1 - 74 channels: Fp1, Fp2, F3 ...>"
+    )
+    assert m10.__repr__() == (
+        "<Montage | easycap-M10 - 61 channels: 1, 2, 3 ...>"
+    )
+
+
+def test_easycap_M1():
+    """Test easycap_M1."""
+    EXPECTED_HEAD_SIZE = 1.  # 0.085
+
+    montage = read_standard_montage('easycap_M1')
+    eeg_loc = np.array([ch['r'] for ch in _get_dig_eeg(montage.dig)])
+
+    ## XXX Thats what I thought it was right. But substracting the center fails
```

Comment:
@agramfort 	I don't understand why when creating [x,y,z] from [1, phi, theta] why is not center in 0. and if its not centered in 0 why when centering the norm is no longer 1.

Reply:
I don't know. I would suggest to do a stupid for loop with all standard montages. Make sure you can pass them to set_montage that it makes sense using fsaverage anatomy. Make sure about thate set_montage guarantees that the DigMontage can be mapped to head (or is already in head).